app/src/main/java/app/revanced/manager/ui/viewmodel/AppSelectorViewModel.kt

```diff
@@ -25,7 +26,7 @@ class AppSelectorViewModel(
     private val pm: PM,
     private val patchBundleRepository: PatchBundleRepository
 ) : ViewModel() {
-    private val inputFile = File(app.cacheDir, "input.apk").also {
+    private val inputFile = File(app.getDir("ephemeral", Context.MODE_PRIVATE), "input.apk").also {
```

Comment:
"ephemeral" is deleted by `Filesystem` every time the app process is restarted. Use another directory.

Reply:
input.apk should be deleted every time the app restarted, shouldn't it?
Is input.apk needed after restarted?

Reply:
We cannot delete the input APK every time the app process is restarted because of [system-initiated process death](https://galex.dev/posts/process-death-is-the-rule-not-the-exception/#users-experience-system-initiated-process-death-a-lot). The process is killed when the app is in the background, but when the user navigates back a new process is launched and state is restored. System-initiated process death would definitely cause problems if the app is killed while the user is in a screen between `AppSelectorScreen` and `PatcherScreen`.

Reply:
OK understood.
I now get what you mean by "process".

Since this is a single file, I think the destination will be `filesDir`. Is that correct?
( `File(app.filesDir, "input.apk")` )

Reply:
Use something like `getDir("app_selector", Context.MODE_PRIVATE)`. I remember having problems when trying to use `filesDir` directly.

Reply:
Isn't it [`dataDir`](https://developer.android.com/reference/android/content/Context#getDataDir()) ?

> Apps should not use this path directly; they should instead use getFilesDir(), getCacheDir(), getDir(java.lang.String, int), or other storage APIs on this class. 

`filesDir` is a standard location to place a file
https://developer.android.com/training/data-storage/app-specific#internal-access-store-files
I replaced it with `File(app.filesDir, "input.apk")` and there were no problems.

Reply:
In that case `filesDir` is fine.